### DISCUSSION TITLE
How can I run "prisma generate" with custom env file?

### DISCUSSION THREAD

**Discussion by tal-rofe** [prisma/prisma#13432] [Q&A] [answered]
By default Prisma will try loading `.env` files. I want to load a one called `.env.development`.
How can I do so?

I tried running: `dotenv -e ./prisma/.env.development -- prisma generate --schema ./prisma/schema.prisma`
Using the `dotenv-cli` package but it does not help

**Comment by janpio** [comment, marked as answer]
Why does it not help? This is exactly how we document it to be used: https://www.prisma.io/docs/guides/development-environment/environment-variables/managing-env-files-and-setting-variables#using-dotenv-cli-via-command-line

**Reply by tal-rofe** [reply]
@janpio 
Is the command I ran correct?

**Reply by janpio** [reply]
It looks close enough to the example from the docs: `dotenv -e .env3 -- npx prisma db pull`
So if you have a `./prisma/.env.development` file and a global `prisma` CLI this should in theory load the env vars from that file, and use them when running `prisma generate` (although that usually does not really need any env vars... so I am wondering what was or was not working for you here).

**Reply by tal-rofe** [reply]
@janpio 
I have an environment variable for setting the database url.
Isn't the URL necessary for the DB initialization?

**Reply by janpio** [reply]
`generate` only generates a Prisma Client, it does not talk to the database.

So my initial question still stands: Why does it not help? What error message or problem are you getting?

**Reply by tal-rofe** [reply]
@janpio 

Well, it does generates it. But when I run my application, I got the following error:
```

  error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:7
   | 
 6 |   provider = "mongodb"
 7 |   url      = env("DATABASE_URL")
```
This happens when my env file is called `.env.development` in the `prisma` folder.
I run the generate command before:|
`"dotenv -e ./prisma/.env.development prisma generate --schema ./prisma/schema.prisma"```

Most important, when I rename the env file to only ".env" then everything works.

**Reply by janpio** [reply]
You need to use `dotenv -e ./prisma/.env.development` as well when you start your application. Setting it for `prisma generate` only loads the file into your environment for that command, not when you later try to run your application that needs the `DATABASE_URL` environment variable.

**Reply by tal-rofe** [reply]
@janpio 
Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
